crates/block-producer/src/block_builder/mod.rs

```diff
@@ -309,21 +309,14 @@ impl TelemetryInjectorExt for ProposedBlock {
             u32::try_from(self.created_nullifiers().len())
                 .expect("should have less than u32::MAX created nullifiers"),
         );
-        let num_block_created_notes = self
-            .output_note_batches()
-            .iter()
-            .fold(0, |acc, output_notes| acc + output_notes.len());
+        let num_block_created_notes = self.batches().num_created_notes();
         span.set_attribute(
             "block.output_notes.count",
             u32::try_from(num_block_created_notes)
                 .expect("should have less than u32::MAX output notes"),
         );
 
-        let num_batch_created_notes = self
-            .batches()
-            .as_slice()
-            .iter()
-            .fold(0, |acc, batch| acc + batch.output_notes().len());
+        let num_batch_created_notes = self.batches().num_created_notes();
```

Comment:
Is this right? Seems like there is no difference and so `num_erased_notes` will always be 0

Reply:
Ah yeah nice catch. It looks like the `block` variant is incorrect now.

`ProposedBlock::output_note_batches` should be used there it seems, though the naming is a bit confusing :)

I'll create an issue.

Reply:
Nice catch!